src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -1297,6 +1311,30 @@ private BoundExpression RewriteDecimalConversion(BoundConversion oldNode, CSharp
             }
         }
 
+        private BoundExpression RewriteMethodGroupConversion(BoundConversion oldNode, CSharpSyntaxNode syntax, BoundExpression operand, MethodSymbol targetMethod, bool isExtensionMethod, TypeSymbol type)
+        {
+            Debug.Assert((object)targetMethod != null);
+            Debug.Assert(type is NamedTypeSymbol);
+
+            var targetMethodDefinition = targetMethod.OriginalDefinition;
+            var cacheFrame = _methodGroupConversionCacheFrameManager.ObtainCacheFrame(targetMethodDefinition);
+            var typeArguments = MethodGroupConversionCacheFrame.GetTypeArgumentsFromConversion(type, targetMethod);
+
+            var orgSyntax = _factory.Syntax;
+            _factory.Syntax = syntax;
+
+            var boundCacheField = _factory.Field(null, cacheFrame.DelegateField.AsMember(cacheFrame.Construct(typeArguments)));
+            var boundDelegateCreation = new BoundDelegateCreationExpression(syntax, operand, targetMethod, isExtensionMethod, type)
```

Comment:
Same question above. In short, I don't know, and I would like someone from the team to explain.